contracts/utils/Votes.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./Context.sol";
+import "./Counters.sol";
+import "./Checkpoints.sol";
+import "./cryptography/draft-EIP712.sol";
+
+/**
+ * @dev Voting operations.
+ */
+abstract contract Votes is Context, EIP712 {
+    using Checkpoints for Checkpoints.History;
+    using Counters for Counters.Counter;
+
+    bytes32 private constant _DELEGATION_TYPEHASH =
+        keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
+    mapping(address => address) _delegation;
+    mapping(address => Checkpoints.History) _userCheckpoints;
```

Comment:
I think this should be renamed to `_delegateCheckpoints`.

Reply:
which one?

Reply:
Fixed `f054441f`